ReactWindows/ReactNative.Shared/UIManager/IUIBlock.cs

```diff
@@ -0,0 +1,17 @@
+﻿using ReactNative.UIManager;
+
+namespace ReactNative.UIManager
+{
+    /// <summary>
+    /// Interface that represents a block to execute on the UI thread.
+    /// Exposes NativeViewHierarchyManager for third party libraries.
+    /// </summary>
+    public interface IUIBlock
+    {
+        /// <summary>
+        /// Executes the block.
+        /// </summary>
+        /// <param name="nativeViewHierarchyManager"></param>
+        void Execute(NativeViewHierarchyManager nativeViewHierarchyManager);
```

Comment:
@rozele Not sure what you mean here. Do you want this method to have a default body or something?